go-controller/pkg/node/gateway_iptables.go

```diff
@@ -438,7 +438,15 @@ func getGatewayIPTRules(service *kapi.Service, svcHasLocalHostNetEndPnt bool) []
 				}
 			}
 		}
-		for _, externalIP := range service.Spec.ExternalIPs {
+		externalIPs := make([]string, 0)
```

Comment:
A small nit, but you could pre-allocate enough space with `make([]string, 0, len(service.Spec.ExternalIPs) + len(service.Status.LoadBalancer.Ingress))`

Reply:
Yup, will do.

Reply:
Done

Reply:
@dcbw I am having second thoughts about this.
If we preallocate, then we'd need to fill the items by index instead of appending, making the code more ugly. Given each service will have just a handful of external ips, I think it won't stress the gc so much to justify the penalty in readability.

Reply:
@fedepaol nah, you don't need to fill by index, here we're just setting the pre-allocated capacity with the 3rd arg to make(), but you're not setting the actual length (2nd arg). Only if we set the length will it actually allocate the elements out of the capacity and require setting by index.

Reply:
Oh, I totally missed that (actually, I did not notice the 3rd parameter and it's a feature I never used)! Let me fix it, thanks!

Reply:
done